wdtk-datamodel/src/main/java/org/wikidata/wdtk/datamodel/implementation/SnakImpl.java

```diff
@@ -83,12 +95,48 @@ public SnakImpl(PropertyIdValue property) {
 	 */
 	protected SnakImpl(
 			String id,
-			String siteIri) {
+			String siteIri) {	
```

Comment:
So, I tried adding hash as the parameter in its constructor, and it is resulting in changing all the files wherever an object of SnakImpl or its subclass has been created. So, is the correct way to approach it ? 
Also, there is` hashCode` function in ` PropertyIdValueImpl `class

![image](https://user-images.githubusercontent.com/23396490/83325632-5a751d80-a28b-11ea-924a-b6625ba5d59a.png)

Its return type is `int` , so is it the same hashCode?

Reply:
Yes you will need to update calls to this constructor.
For the second question, no, that is the internal hash used in Java (for instance to store snaks as map keys).

Reply:
Ohh, okay. Thank you
So, is it fine if I add hash as a parameter to other functions as well, such as for the following functions in the `DataObjectFactoryImpl.java`

![image](https://user-images.githubusercontent.com/23396490/83326441-a62ac580-a291-11ea-870f-16b2592c62bb.png)

Thank you

Reply:
Yes, in each case we need to decide whether to add a parameter to the function where the constructor is called, such that a non-empty hash can be provided by the caller, or to simply provide a null hash to the constructor.

In the case above, adding parameters to the function makes sense, since this factory class is really there to copy the datamodel classes faithfully. In other contexts, such as the `Datamodel` helper, we probably want to make sure the existing method signatures are maintained, since in most cases users will not want to specify a hash there. We could add other methods which support hashes alongside the default ones.

Reply:
Okay, I believe this to be a long process, and changes need to be done keeping utility of each function in mind, which requires confirmation from your side for each one of them?

Reply:
You can just propose something by adding changes to the PR and we can discuss it together